tests/components/brunt/test_config_flow.py

```diff
@@ -0,0 +1,186 @@
+
+
+async def test_import_duplicate_login(hass):
+    """Test uniqueness of username."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CONFIG,
+        title="test-username",
+        unique_id="test-username",
+    )
+    entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.brunt.config_flow.BruntClientAsync.async_login",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.brunt.async_setup_entry",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=CONFIG
+        )
+        await hass.async_block_till_done()
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+async def test_form_duplicate_login(hass):
+    """Test uniqueness of username."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CONFIG,
+        title="test-username",
+        unique_id="test-username",
+    )
+    entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.brunt.config_flow.BruntClientAsync.async_login",
+        return_value=None,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}, data=CONFIG
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+@pytest.mark.parametrize(
+    "side_effect, error_message",
+    [
+        (ServerDisconnectedError, "cannot_connect"),
+        (ClientResponseError(None, None, status=403), "invalid_auth"),
+        (ClientResponseError(None, None, status=401), "unknown"),
+        (Exception, "unknown"),
+    ],
+)
+async def test_form_error(hass, side_effect, error_message):
+    """Test we handle cannot connect."""
+    with patch(
+        "homeassistant.components.brunt.config_flow.ClientResponseError.__str__",
+        return_value="ClientResponseError",
+    ), patch(
+        "homeassistant.components.brunt.config_flow.BruntClientAsync.async_login",
+        side_effect=side_effect,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}, data=CONFIG
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["errors"] == {"base": error_message}
+
+
+@pytest.mark.parametrize("side_effect", [None, Exception])
+async def test_reauth(hass, side_effect):
+    """Test uniqueness of username."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CONFIG,
+        title="test-username",
+        unique_id="test-username",
+    )
+    entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.brunt.config_flow.BruntClientAsync.async_login",
+        side_effect=ClientResponseError(None, None, status=403),
```

Comment:
Why do we set this side effect? We start the reauth flow explicitly so the error in the integration isn't important at this point.